indexing-service/src/main/java/org/apache/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -257,7 +257,7 @@ public void onFailure(Throwable throwable)
                         }
                       }
                     },
-                    // The callback is non-blocking and quick, so it's OK to schedule it using directExecutor()
+                    // The callback is mostly non-blocking and quick, so it's OK to schedule it using directExecutor()
```

Comment:
[In my comment](https://github.com/apache/incubator-druid/pull/8809#discussion_r342940649), I wrote "*you may argue that*... this code is mostly non-blocking", I didn't *state* that (and I actually don't know). I. e. the fact that the synchronized block in `onSuccess()` requires some proof, expressed in a comment. There is not enough information to make such proof (or disproof) just in this code excerpt in Github's interface. A wider context (specifically, the code which synchronizes on `waitingForMonitor` and waits) should be considered.

Reply:
get it. the synchronization of waitingForMonitor involves a while loop. Will use dedicated thread